hledger-lib/Hledger/Utils.hs

```diff
@@ -186,12 +184,7 @@ readFileOrStdinPortably f = openFileOrStdin f ReadMode >>= readHandlePortably
     openFileOrStdin f m   = openFile f m
 
 readHandlePortably :: Handle -> IO Text
-readHandlePortably h = do
-  hSetNewlineMode h universalNewlineMode
-  menc <- hGetEncoding h
-  when (fmap show menc == Just "UTF-8") $  -- XXX no Eq instance, rely on Show
-    hSetEncoding h utf8_bom
-  T.hGetContents h
+readHandlePortably h = hSetNewlineMode h universalNewlineMode *> TIO.hGetContents h
```

Comment:
Doesn't this read with the system locale's encoding as before ? Shouldn't we be enforcing utf8 now ?

Reply:
I believe it does. `openFile` comes from `with-utf8` (I'll qualify the import so that's more obvious), and `stdin` should be taken care of by the `withUtf8` wrapper around `main`.